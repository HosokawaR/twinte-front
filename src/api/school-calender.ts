import axios, { AxiosResponse } from 'axios'

import { BASE_URL } from '../common/config'

export type SchoolCalender = {
  event?: {
    description: string
    event_type: string
    metadata: any
    date: string
  }
  substituteDay?: {
    date: string
    change_to: string
  }
  module: string
}

export const getSchoolCalender = async (
  date: string
): Promise<AxiosResponse<SchoolCalender>> => {
  const url = `${BASE_URL}/school-calender​/${date}`
    // ゼロ幅スペース削除
    .replace(/[\u200B-\u200D\uFEFF]/g, '')

  return axios.get<SchoolCalender>(url)
}

export type SubstitutesList = {
  date: string
  change_to: string
}[]

export const getSchoolCalenderSubstitutesList = async (): Promise<
  AxiosResponse<SubstitutesList>
> => {
  const url = `${BASE_URL}/school-calender/substitutes/list`

  return axios.get<SubstitutesList>(url)
}
