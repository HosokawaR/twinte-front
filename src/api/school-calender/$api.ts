/* eslint-disable */
import { AspidaClient, BasicHeaders, dataToURLString } from 'aspida'
import { Methods as Methods0 } from './_date@string'
import { Methods as Methods1 } from './substitutes/list'

const api = <T>({ baseURL, fetch }: AspidaClient<T>) => {
  const prefix = (baseURL === undefined ? '/v1/' : baseURL).replace(/\/$/, '')
  const PATH0 = '/school-calender'
  const PATH1 = '/school-calender/substitutes/list'
  const GET = 'GET'

  return {
    _date: (val0: string) => {
      const prefix0 = `${PATH0}/${val0}`

      return {
        get: (option?: { config?: T }) =>
          fetch<
            Methods0['get']['resBody'],
            BasicHeaders,
            Methods0['get']['status']
          >(prefix, prefix0, GET, option).json(),
        $get: (option?: { config?: T }) =>
          fetch<
            Methods0['get']['resBody'],
            BasicHeaders,
            Methods0['get']['status']
          >(prefix, prefix0, GET, option)
            .json()
            .then((r) => r.body),
        $path: () => `${prefix}${prefix0}`,
      }
    },
    substitutes: {
      list: {
        get: (option: { query: Methods1['get']['query']; config?: T }) =>
          fetch<
            Methods1['get']['resBody'],
            BasicHeaders,
            Methods1['get']['status']
          >(prefix, PATH1, GET, option).json(),
        $get: (option: { query: Methods1['get']['query']; config?: T }) =>
          fetch<
            Methods1['get']['resBody'],
            BasicHeaders,
            Methods1['get']['status']
          >(prefix, PATH1, GET, option)
            .json()
            .then((r) => r.body),
        $path: (option?: { method?: 'get'; query: Methods1['get']['query'] }) =>
          `${prefix}${PATH1}${
            option && option.query ? `?${dataToURLString(option.query)}` : ''
          }`,
      },
    },
  }
}

export type ApiInstance = ReturnType<typeof api>
export default api
