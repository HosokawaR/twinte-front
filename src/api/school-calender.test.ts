import { getSchoolCalender } from './school-calender'

jest.setTimeout(30000)

describe(getSchoolCalender.name, () => {
  it('basic usage', async () => {
    const cal = await getSchoolCalender('2020-06-01')
    expect(cal.status).toBe(200)
    expect(cal.data).toMatchSnapshot('cal__result')
  })
})
