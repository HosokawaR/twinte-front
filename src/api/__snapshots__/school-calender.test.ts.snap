// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic usage: cal__result 1`] = `
Object {
  "module": "æ˜¥B",
}
`;
